test/e2e/installplan_e2e_test.go

```diff
@@ -60,6 +63,140 @@ var _ = Describe("Install Plan", func() {
 		TearDown(testNamespace)
 	})
 
+	When("an InstallPlan step contains a deprecated resource version", func() {
+		var (
+			pdb      policyv1beta1.PodDisruptionBudget
+			manifest string
+			counter  float64
+		)
+
+		BeforeEach(func() {
+			dc, err := discovery.NewDiscoveryClientForConfig(ctx.Ctx().RESTConfig())
+			Expect(err).ToNot(HaveOccurred())
+
+			v, err := dc.ServerVersion()
+			Expect(err).ToNot(HaveOccurred())
+
+			if minor, err := strconv.Atoi(v.Minor); err == nil && minor < 21 {
+				// This is a tactical can-kick with
+				// the expectation that the
+				// event-emitting behavior being
+				// tested in this context will have
+				// moved by the time 1.25 arrives.
+				Skip("hack: test is dependent on 1.21+ behavior")
+			}
+		})
+
+		BeforeEach(func() {
+			counter = 0
+			for _, metric := range getMetricsFromPod(ctx.Ctx().KubeClient(), getPodWithLabel(ctx.Ctx().KubeClient(), "app=catalog-operator"), "8081") {
+				if metric.Family == "installplan_warnings_total" {
+					counter = metric.Value
+				}
+			}
+
+			csv := newCSV("test-csv", testNamespace, "", semver.Version{}, nil, nil, nil)
+			Expect(ctx.Ctx().Client().Create(context.TODO(), &csv)).To(Succeed())
+
+			pdb = policyv1beta1.PodDisruptionBudget{
```

Comment:
using a PDB makes sense. I'm wondering if it's worth adding a case for v1beta1 CRDs (since it's really what we care about) or if we are confident enough in the new codepath with a PDB in the installplan only? 

CRDs have some additional validation logic when applying to a cluster with the CRD already present -- stuff like schema validation, etc. If that fails, I don't think the alert would fire, which I believe is what we want.